controlplane/internal/controller/agentcontrolplane_controller.go

```diff
@@ -93,12 +96,26 @@ func (r *AgentControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		}
 		return ctrl.Result{}, err
 	}
+	log.Info("Started reconciling AgentControlPlane")
 	defer func() {
 		if rerr = r.Client.Status().Update(ctx, acp); rerr != nil {
 			log.Error(rerr, "couldn't update AgentControlPlane Status", "name", acp.Name, "namespace", acp.Namespace)
 		}
+		log.Info("Finished reconciling AgentControlPlane")
 	}()
 
+	if acp.DeletionTimestamp != nil {
+		log.Info("ACP is being deleted")
+		return r.handleDeletion(ctx, acp)
+	} else if !controllerutil.ContainsFinalizer(acp, ACP_FINALIZER) {
```

Comment:
else is not necessary

Reply:
How come? This only updates the CR if it doesn't already have the finalizer (we don't want it to update all the time), otherwise it skips the update

Reply:
if `DeletionTimestamp` is not nil, we will handle deletion and return.
This means we can remove the else, we can just have another if condition (`if !controllerutil.ContainsFinalizer(...)`).
I think this would make the code more readable.

Reply:
Oh I got you, that makes sense, thanks for explaining. Modified as suggested!